golem/managers/proposal/plugins/negotiating/negotiating_plugin.py

```diff
@@ -40,24 +41,43 @@ async def get_proposal(self) -> Proposal:
         while True:
             proposal = await self._get_proposal()
             demand_data = await proposal.demand.get_demand_data()
+            provider_name = await proposal.get_provider_name()
 
             try:
                 negotiated_proposal = await self._negotiate_proposal(demand_data, proposal)
-                self._success_count += 1
-                logger.info(
-                    f"Negotiation based on proposal `{proposal}` succeeded"
+            except ProposalRejected as e:
+                self._fail_count += 1
+                logger.debug(
+                    f"Negotiation based on proposal `{proposal}` from `{provider_name}` failed as"
+                    f" it was rejected by the provider: `{e}`, retrying with new one..."
                     "\nsuccess count: "
-                    f"{self._success_count}/{self._success_count + self._fail_count}"
+                    f"{self._success_count}/{self._success_count + self._fail_count}",
+                )
+            except RejectProposal as e:
+                self._fail_count += 1
+                logger.debug(
+                    f"Negotiation based on proposal `{proposal}` from `{provider_name}` failed as"
+                    f" it was rejected by requestor: `{e}`, retrying with new one..."
+                    "\nsuccess count: "
+                    f"{self._success_count}/{self._success_count + self._fail_count}",
                 )
-                return negotiated_proposal
             except Exception:
                 self._fail_count += 1
                 logger.debug(
-                    f"Negotiation based on proposal `{proposal}` failed, retrying with new one..."
+                    f"Negotiation based on proposal `{proposal}` from `{provider_name}` failed,"
```

Comment:
this part could be constructed above and just reused, otherwise, the code looks awfully repetitive

Reply:
actually, this could be just one `except` which varies the message depending on the class of the exception...

Reply:
Well... I can't argue, good point. But I'll optimize true/false branches separately, further optimization does not make sense.